Wikipedia/Code/PreviewAndSaveViewController.m

```diff
@@ -501,43 +501,25 @@ - (void)fetchFinished:(id)sender
             } break;
         }
     } else if ([sender isKindOfClass:[EditTokenFetcher class]]) {
-        EditTokenFetcher *tokenFetcher = (EditTokenFetcher *)sender;
-
-        void (^upload)() = ^void() {
-            NSMutableDictionary *editTokens = self.keychainCredentials.editTokens;
-            NSString *editToken = editTokens[tokenFetcher.articleURL.wmf_language];
-            (void)[[WikiTextSectionUploader alloc] initAndUploadWikiText:tokenFetcher.wikiText
-                                                           forArticleURL:tokenFetcher.articleURL
-                                                                 section:tokenFetcher.section
-                                                                 summary:tokenFetcher.summary
-                                                               captchaId:tokenFetcher.captchaId
-                                                             captchaWord:tokenFetcher.captchaWord
-                                                                   token:editToken
-                                                             withManager:[QueuesSingleton sharedInstance].sectionWikiTextUploadManager
-                                                      thenNotifyDelegate:self];
-        };
-
         switch (status) {
             case FETCH_FINAL_STATUS_SUCCEEDED: {
-                NSMutableDictionary *editTokens =
-                    self.keychainCredentials.editTokens;
-                NSString *domain = self.section.url.wmf_language;
-                if (domain && tokenFetcher.token) {
-                    editTokens[domain] = tokenFetcher.token;
-                    self.keychainCredentials.editTokens = editTokens;
-                }
-                upload();
+                EditTokenFetcher *tokenFetcher = (EditTokenFetcher *)sender;
+
+                (void)[[WikiTextSectionUploader alloc] initAndUploadWikiText:tokenFetcher.wikiText
```

Comment:
seems like the `WikiTextSectionUploader` (and the `EditTokenFetcher` initialized on line 675) should be assigned to properties and retained. Having unowned objects like this could lead to them being dealloc'd and not actually completing the fetch.